libpod/healthcheck.go

```diff
@@ -0,0 +1,62 @@
+package libpod
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+// HealthCheckStats represents the current state of a container
+type HealthCheckStatus int
+
+const (
+	HealthCheckSuccess           HealthCheckStatus = 0
+	HealthCheckFailure           HealthCheckStatus = 1
+	HealthCheckContainerStopped  HealthCheckStatus = 99
+	HealthCheckContainerNotFound HealthCheckStatus = 999
+	HealthCheckNotDefined        HealthCheckStatus = 9999
+)
+
+type HealthCheck struct {
+	interval    int //(default: 30s)
+	timeout     int //(default: 30s)
+	startPeriod int //(default: 0s)
+	retries     int //(default: 3)
+	command     string
+}
+
+// HealthCheck executes the ...
+func (r *Runtime) HealthCheck(name string) (HealthCheckStatus, error) {
+	container, err := r.LookupContainer(name)
+	if err != nil {
+		return HealthCheckContainerNotFound, errors.Wrapf(err, "unable to lookup %s to perform a health check", name)
+	}
+	cstate, err := container.State()
+	if err != nil {
+		return -1, err
+	}
+	if cstate != ContainerStateRunning {
+		return HealthCheckContainerStopped, errors.Errorf("container %s is not running", container.ID())
+	}
+	if !container.HasHealthCheck() {
+		return HealthCheckNotDefined, errors.Errorf("container %s has no defined healthcheck", name)
+	}
+	return container.RunHealthCheck()
+}
+
+// RunHealthCheck runs the health check as defined by the container
+func (c *Container) RunHealthCheck() (HealthCheckStatus, error) {
+	var newCommand []string
+	hcCommand := c.HealthCheckConfig().Test
```

Comment:
This is unsafe - need to check if HealthCheckConfig() returned nil before doing any deferences